### PR TITLE
Improve exception messages

### PR DESCRIPTION
**Pull request by ueman**
Further improve exception messages.

### New Pull Request Checklist

- [x] I have read the [Documentation](https://pub.dev/documentation/dio/latest/)
- [x] I have searched for a similar pull request in the [project](https://github.com/cfug/dio/pulls) and found none
- [x] I have updated this branch with the latest `main` branch to avoid conflicts (via merge from master or rebase)
- [x] I have added the required tests to prove the fix/feature I'm adding
- [x] I have updated the documentation (if necessary)
- [x] I have run the tests without failures
- [x] I have updated the `CHANGELOG.md` in the corresponding package

### CODE REVIEW

File: dio/lib/src/dio_exception.dart [hunk 1]
```diff
@@ -96,24 +96,32 @@ class DioException implements Exception {
...
-      DioException(
-        type: DioExceptionType.connectionTimeout,
-        message: 'The request connection took '
-            'longer than $timeout. It was aborted.',
-        requestOptions: requestOptions,
-        response: null,
-        error: error,
-      );
+  }) {
+    final message = 'The request connection took longer than $timeout '
```
Comment (AlexV525) [inline dio/lib/src/dio_exception.dart:100]: Make this inline?

File: dio/lib/src/dio_exception.dart [hunk 2]
```diff
@@ -154,7 +165,8 @@ class DioException implements Exception {
   }) =>
       DioException(
         type: DioExceptionType.connectionError,
-        message: 'The connection errored: $reason',
+        message: 'The connection errored: $reason. '
+            'This indicates an error which most likely can\'t be solved with DIO.',
```
Comment (AlexV525) [inline dio/lib/src/dio_exception.dart:169]: Correct me if I'm wrong, I'm not sure about *with* or *by*

```suggestion
        message: 'The connection errored: $reason. '
            'This indicates an error which most likely cannot be solved by the library.',
```

**Review by AlexV525 — APPROVED** [review]
LGTM with nits

File: dio/lib/src/dio_exception.dart (hunk 2 above)
Comment (ueman) [reply dio/lib/src/dio_exception.dart:169]: > I'm not sure about with or by

Me neither, but `by` reads better, so I go with it.

File: dio/lib/src/dio_exception.dart (hunk 1 above)
Comment (ueman) [reply dio/lib/src/dio_exception.dart]: Done

**Comment by ueman** [comment]
I haven't added a changelog, since these changes already covered by the changelog entry from the previous PR.

**Pull request merged**